crates/test-collector/src/lib.rs

```diff
@@ -556,16 +557,31 @@ pub fn collect_tests(
         })
         .collect_vec()
         .into_iter()
-        .map(|(test_name, config)| TestCase {
-            name: test_name,
-            available_gas: config.available_gas,
-            ignored: config.ignored,
-            expected_result: config.expected_result,
-            fork_config: config.fork_config,
-            fuzzer_config: config.fuzzer_config,
+        .map(|(test_name, config)| {
+            if config.available_gas.is_some() {
+                gas_error = Some(Err(
+                    anyhow!
+                        ("{} - Gas calculation used for available_gas is incorrect, because it doesn't include the cost of contract execution.",
+                         test_name
+                    )
+                ));
+
+            };
+            TestCase {
+                name: test_name,
+                available_gas: config.available_gas,
+                ignored: config.ignored,
+                expected_result: config.expected_result,
+                fork_config: config.fork_config,
+                fuzzer_config: config.fuzzer_config,
+            }
         })
         .collect();
 
+    if let Some(err) = gas_error {
+        return err;
+    }
```

Comment:
Instead of using approach like this, lets refactor the use of `map` so it returns a Result which we can then bubble.

Change `collect()` to something along the lines of this `collect::<Result<Vec<_>>>()?` and update the code inside the map so it users `Result` there.

Reply:
Doesn't `bail!` solve all of this and is also more readable?

Reply:
bail! is only shortcut of `return new Err(anyhow!(..))`. @cptartur  I have no idea how to achive, collect::<Result<Vec<_>>>()?. If I use bail! I will get `Vec<Result<_>>`

Reply:
Yep, thus returning from the function and bubbling the error. If we do it above where I suggested it all this mapping won't be necessary

Reply:
Oh nvm, it is inside a `map` 🤦 Then I guess this approach is fine